src/Symfony/Component/Form/ButtonBuilder.php

```diff
@@ -503,6 +509,28 @@ public function setAutoInitialize($initialize)
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function setPosition($position)
+    {
+        if ($this->locked) {
+            throw new BadMethodCallException('The config builder cannot be modified anymore.');
+        }
+
+        if (is_string($position) && ($position !== 'first') && ($position !== 'last')) {
+            throw new InvalidConfigurationException('If you use position as string, you can only use "first" & "last".');
+        }
+
+        if (is_array($position) && !isset($position['before']) && !isset($position['after'])) {
+            throw new InvalidConfigurationException('If you use position as array, you must at least define the "before" or "after" option.');
```

Comment:
`When using position as an array, the "before" or "after" option must be defined.`